From d0793693ca877b8614d64bcd70051351df4f6c44 Mon Sep 17 00:00:00 2001
From: Magnus Damm <damm@opensource.se>
Date: Sat, 25 Feb 2012 22:14:18 +0100
Subject: PM / Domains: Fix include for PM_GENERIC_DOMAINS=n case

Fix pm_genpd_init() arguments and make sure dev_gpd_data() and
simple_qos_governor exist regardless of CONFIG_PM_GENERIC_DOMAINS
setting.

Signed-off-by: Magnus Damm <damm@opensource.se>
Signed-off-by: Rafael J. Wysocki <rjw@sisk.pl>
(cherry picked from commit b642631d38c28fefd1232a6b96713eb54b60130d)

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 include/linux/pm_domain.h |   12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/include/linux/pm_domain.h b/include/linux/pm_domain.h
index e76cc9a..5c2bbc2 100644
--- a/include/linux/pm_domain.h
+++ b/include/linux/pm_domain.h
@@ -155,6 +155,10 @@ extern bool default_stop_ok(struct device *dev);
 extern struct dev_power_governor pm_domain_always_on_gov;
 #else
 
+static inline struct generic_pm_domain_data *dev_gpd_data(struct device *dev)
+{
+	return ERR_PTR(-ENOSYS);
+}
 static inline struct generic_pm_domain *dev_to_genpd(struct device *dev)
 {
 	return ERR_PTR(-ENOSYS);
@@ -195,7 +199,8 @@ static inline int __pm_genpd_remove_callbacks(struct device *dev, bool clear_td)
 {
 	return -ENOSYS;
 }
-static inline void pm_genpd_init(struct generic_pm_domain *genpd, bool is_off)
+static inline void pm_genpd_init(struct generic_pm_domain *genpd,
+				 struct dev_power_governor *gov, bool is_off)
 {
 }
 static inline int pm_genpd_poweron(struct generic_pm_domain *genpd)
@@ -206,11 +211,8 @@ static inline bool default_stop_ok(struct device *dev)
 {
 	return false;
 }
+#define simple_qos_governor NULL
 #define pm_domain_always_on_gov NULL
-static inline struct generic_pm_domain_data *dev_gpd_data(struct device *dev)
-{
-	return NULL;
-}
 #endif
 
 static inline int pm_genpd_remove_callbacks(struct device *dev)
-- 
1.7.10.1.362.g242cab3

